src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,6 +82,34 @@ const getRelationParts = (val: unknown): string[] => {
   return []
 }
 
+//Placeholder method - this will be fleshed out in AJ-790, but is needed now for WDSTroubleshooter to work
+export const resolveWdsApp = apps => {
```

Comment:
Do you want to bump the values/parse out the URL of where we call upon `proxyUrlPromise` now that this is returning the entire object (not just the URL)? Unless you were anticipating that I covered that in my current PR?

Reply:
Not sure I understand exactly what you're suggesting here, but I didn't put much effort into this method since I was anticipating you might get it done first.  Although it has occurred to me that we're actually looking for slightly different things (since I'm going to list # of leo apps and app status (e.g. RUNNING, PROVISIONING) and you're looking for exactly one app that is definitely RUNNING) maybe I should make my own method... there's just going to be a decent amount of overlap.

Reply:
+1 to combining these functions

Reply:
Perhaps we could just have a parent function though that calls the API once, then passes the payload to either this function to resolve the URL, or to another function that translates the dictionary to be rendered by the Troubleshooter?

Reply:
I think that's basically what we're doing - `getV2AppInfo` calls the API and gets the apps, then we use either your function - `resolveWdsUrl` or mine - `resolveWdsApp`.

Reply:
I agree that `resolveWdsApp` and `getWdsUrl`/`resolveWdsUrl` should be combined. If we need both because the former returns the full app definition and the latter returns just the url, let's put all the business logic into `resolveWdsApp`and change `resolveWdsUrl` to delegate, e.g.:
```
export const resolveWdsUrl = apps => {
  return resolveWdsApp(apps)?.proxyUrls.wds
}
```
(untested, but hopefully shows what I'm suggesting)

Reply:
Should we be concerned that  the combined `resolveWdsApp` will end up creating a new app if one is missing, whether it's called from the troubleshooter or the rest of the page?  

For getting the proxyUrl for normal operation we want to make sure we get the correct RUNNING app; for the troubleshooter it would be ok to get one that's PROVISIONING - the information about what non-running app is out there might be useful to have for the troubleshooter.  I would propose having `resolveWdsApp` be less strict about what it returns, then `resolveWdsUrl` would check that the result is running before grabbing the proxyUrl.

Reply:
FYI @aaronkanzer I've pushed an updated version of `resolveWdsUrl` that just pulls the url from the `resolveWdsApp` result (if it's running) - so if this gets done before your PR make sure you check the diffs carefully!

Reply:
sounds good! thanks for doing that 🎉

Reply:
Just one concern -- do the `const proxyUrl = await this.proxyUrlPromise` calls still work as intended?

I know our unit tests are a bit shaky sometimes ha 😰

Reply:
They seem to!  When the app is actually running properly everything works and responds as intended